containers/codespaces-linux/.devcontainer/base.Dockerfile

```diff
@@ -40,6 +40,9 @@ RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
     && apt-get -y install build-essential cmake cppcheck valgrind clang lldb llvm gdb \
     # Install tools and shells not in common script
     && apt-get install -yq vim vim-doc xtail software-properties-common libsecret-1-dev \
+    # Install additional tools (useful for 'puppeteer' project)
+    && apt-get install -y --no-install-recommends libnss3 libnspr4 libatk-bridge2.0-0 libatk1.0-0 libx11-6 libpangocairo-1.0-0 \
```

Comment:
Is there any sort of unit test we can add to this repo to verify that puppeteer works in the built image? I think we have similar tests for other tools. That would be a good regression test to ensure puppeteer continues to work well as the image changes over time.

Reply:
thats a great idea.  If some testing framework doesn't already exist it would be a great thing to add/i'd be happy to look into it

Reply:
Check out https://github.com/microsoft/vscode-dev-containers/blob/master/containers/codespaces-linux/test-project/test.sh. You can run the tests locally, and I think they also run in checks/CI.

Reply:
Yep this run every other day from this workflow: https://github.com/microsoft/vscode-remote-containers/actions/runs/467541913

Runs "test-project/test.sh" after using Remote - Containers to spin up the container. Also useful way to detect regressions in the product work flow.  Even without it, it verifies the config is successful.

We can probably separate out codespaces to make it more visible right now its in the `regression-test-definitions (4, 14)` job.  There's enough of them that I spin up 16 runs concurrently to get it to complete in a reasonable time.

 There's one regression error in a community contributed one right now (azure-blockchain) that needs a look, but everything else is green right now.

Reply:
oh sweet - didn't realize those scripts got called by CI.  Will make the changes 👍